### PR TITLE
Add patoolib hook

### LINKED ISSUE #3013: Patoolib integration with Pyinstaller
Hi!

I'm using patoolib in one of my projects for extracting archives. I used Pyinstaller to get a single executable and the it does't work. My MWE is the following:

```
from patoolib import extract_archive
extract_archive('Test.zip',verbosity=-1)
```

Where "Test.zip" only contains a "Test.txt" inside. The executable was built with:

```
pyinstaller .\Zip.py -F
```

Running the script directly works, however with the executable I get the following error:

> Traceback (most recent call last):
>   File "site-packages\patoolib\__init__.py", line 553, in get_archive_cmdlist_func
>   File "importlib\__init__.py", line 126, in import_module
>   File "<frozen importlib._bootstrap>", line 994, in _gcd_import
>   File "<frozen importlib._bootstrap>", line 971, in _find_and_load
>   File "<frozen importlib._bootstrap>", line 941, in _find_and_load_unlocked
>   File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
>   File "<frozen importlib._bootstrap>", line 994, in _gcd_import
>   File "<frozen importlib._bootstrap>", line 971, in _find_and_load
>   File "<frozen importlib._bootstrap>", line 953, in _find_and_load_unlocked
> ModuleNotFoundError: No module named 'patoolib.programs'
> 
> During handling of the above exception, another exception occurred:
> 
> Traceback (most recent call last):
>   File "Zip.py", line 3, in <module>
>     extract_archive('Test.zip',verbosity=-1)
>   File "site-packages\patoolib\__init__.py", line 684, in extract_archive
>   File "site-packages\patoolib\__init__.py", line 472, in _extract_archive
>   File "site-packages\patoolib\__init__.py", line 555, in get_archive_cmdlist_func
> patoolib.util.PatoolError: No module named 'patoolib.programs'
> [3748] Failed to execute script Zip

I'm guessing it has to do with the way Pyinstaller incorporates patoolib into the executable, since the script runs nicely.

I'm using Windows 10 x64, with Python 3.6.3 x64 and Pyinstaller 3.3. Any questions I may answer, let me know. However, I believe the MWE can reproduce the issue quite well.

Thank you!

### PR DESCRIPTION
**Pull request by mtayeldev**
Fixes #3013

Updated the list with a recent(ish) list of modules.

### CODE REVIEW

File: PyInstaller/hooks/hook-patoolib.py [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+# The full license is in the file COPYING.txt, distributed with this software.
+#-----------------------------------------------------------------------------
+
+
+"""
+patoolib uses importlib and pyinstaller doesn't find it and add it to the list of needed modules
+"""
+
+
+hiddenimports = [
```
Comment (rokm) [inline PyInstaller/hooks/hook-patoolib.py]: Looks like this should be a

```
from PyInstaller.utils.hooks import collect_submodules
hiddenimports = collect_submodules('patoolib.programs')
```

**Comment by rokm** [comment]
Actually, sorry for not catching this earlier, but this hook should go into https://github.com/pyinstaller/pyinstaller-hooks-contrib, not here.

**Comment by rokm** [comment]
So please open a PR with the hook there, and be sure to add a news fragment: https://github.com/pyinstaller/pyinstaller-hooks-contrib/blob/master/news/README.txt

**Comment by mtayeldev** [comment]
> So please open a PR with the hook there, and be sure to add a news fragment: https://github.com/pyinstaller/pyinstaller-hooks-contrib/blob/master/news/README.txt

didn't know that existed 👍

**Comment by mtayeldev** [comment]
replaced by https://github.com/pyinstaller/pyinstaller-hooks-contrib/pull/748

**Pull request closed without merging**